packages/esm-sms-app/src/providers/providers-overview/providers-overview.component.tsx

```diff
@@ -0,0 +1,312 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { CardHeader, ErrorState } from '@openmrs/esm-patient-common-lib';
+import { useProviderConfigurations } from '../../hooks/useProviderConfigurations';
+import { useConfigurationsSorting } from '../utils';
+import {
+  Button,
+  InlineLoading,
+  DataTable,
+  DataTableSkeleton,
+  Pagination,
+  TableContainer,
+  Table,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableExpandHeader,
+  Tile,
+} from '@carbon/react';
+import { Add } from '@carbon/react/icons';
+import styles from './providers-overview.scss';
+import {
+  useConfig,
+  usePagination,
+  useLayoutType,
+  isDesktop as isDesktopLayout,
+  showModal,
+  ExtensionSlot,
+} from '@openmrs/esm-framework';
+import { type ConfigObject } from '../../config-schema';
+import { ConfigurationsActionMenu } from './providers-action-menu.component';
+import { TableExpandRow } from '@carbon/react';
+import { TableExpandedRow } from '@carbon/react';
+import { launchOverlay } from '../../hooks/useOverlay';
+import AddProviderConfigForm from '../add-config-form/provider-config-form.component';
+import { EmptyState } from '../empty-state/empty-state.component';
+import { Tag } from '@carbon/react';
+import classNames from 'classnames';
+
+const ProvidersListTable = () => {
+  const { t } = useTranslation();
+  const config = useConfig<ConfigObject>();
+  const layout = useLayoutType();
+  const isDesktop = isDesktopLayout(layout);
+  const isTablet = !isDesktop;
+  const pageSizes = config?.configurationPageSize ?? [10, 20, 30, 40, 50];
+  const [pageSize, setPageSize] = useState(config?.configurationPageSize ?? 10);
```

Comment:
Line 48 assumes `configurationPageSize` is an array, while line 49 assumes it is a number. It can only be one. As stated before I think you should do away with the configuration option altogether though.

Also `config` will never be nully. The optional chaining is not needed.